File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Ask user to input card number
    long _default1 = 1000000000000;
    long _default2 = 1000000000000000;
    long number;

    do
    {
        number = get_long("Number: ");
    }
    while (!number || number < 0);

    long _number = number;

    // Check sum
    int position = 0;
    int total = 0;

    while (number)
    {
        position++;

        if (position % 2 == 0)
        {
            int mod = number % 10;

            if (mod * 2 / 10 > 0)
            {
                total += ((mod * 2 / 10) + (mod * 2 % 10));
            }
            else
            {
                total += (number % 10) * 2;
            }
        }
        else
        {
            total += number % 10;
        }
        number /= 10;
    }

    if (position < 13 || position > 16 || total % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        long _count = 1;

        for (int j = 1; j < position - 1; j++)
        {
            _count *= 10;
        }

        if (position == 15 && (_number / _count == 34 || _number / _count == 37))
        {
            printf("AMEX\n");
        }
        else if ((position == 13 || position == 16) && ((_number / _count) / 10 == 4))
        {
            printf("VISA\n");
        }
        else if (position == 16 && (_number / _count == 51 || _number / _count == 52 || _number / _count == 53 ||
                                    _number / _count == 54 || _number / _count == 55))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
}

File: sentimental-credit/credit.py
# TODO
from cs50 import get_int


def main():
    number = -1
    while number < 0:
        number = get_int("Number: ")
    _number = number
    position = 0
    total = 0
    while number:
        position += 1
        if int(position % 2) == 0:
            mod = int(number % 10)
            if int(mod * 2 / 10) > 0:
                total += int(mod * 2 / 10) + int(mod * 2 % 10)
            else:
                total += int(number % 10) * 2
        else:
            total += int(number % 10)
        number = int(number / 10)
    if position < 13 or position > 16 or total % 10 != 0:
        print("INVALID")
    else:
        _count = 1
        for j in range(position - 2):
            _count *= 10
        if position == 15 and (
            int(_number / _count) == 34 or int(_number / _count) == 37
        ):
            print("AMEX")
        elif (position == 13 or position == 16) and (
            int(int(_number / _count) / 10) == 4
        ):
            print("VISA")
        elif position == 16 and (
            int(_number / _count) == 51
            or int(_number / _count) == 52
            or int(_number / _count) == 53
            or int(_number / _count) == 54
            or int(_number / _count) == 55
        ):
            print("MASTERCARD")
        else:
            print("INVALID")


main()
